diff --git a/Source/Urho3D/UI/CheckBox.h b/Source/Urho3D/UI/CheckBox.h
index 11ed6895f..e1ecbdee6 100644
--- a/Source/Urho3D/UI/CheckBox.h
+++ b/Source/Urho3D/UI/CheckBox.h
@@ -50,6 +50,8 @@ public:
 
     /// Set checked state.
     void SetChecked(bool enable);
+    /// Set checked state without generating an event.
+    void SetCheckedInternal(bool enable){ checked_ = enable; }
     /// Set checked image offset.
     void SetCheckedOffset(const IntVector2& offset);
     /// Set checked image offset.
